Automattic/wp-calypso
# Add Jetpack Cloud Backup Download Flow

#### Changes proposed in this Pull Request

* Fill in download flow with unstyled flow
* Hook up "Download backup" button to above flow
* Some more selector work, including adding tracking for the "download progress" request to hold rendering the page until we have the info we need

1. `/backups/:siteSlug` start download
<img width="600" alt="start download" src="https://user-images.githubusercontent.com/2810519/76254308-7b3c5280-6209-11ea-97ed-f1cb893313d3.png">

2. `/backups/:siteSlug/download/:rewindId` download detail

a. confirm
<img width="600" alt="download detail confirm" src="https://user-images.githubusercontent.com/2810519/76254560-e6862480-6209-11ea-960d-5e2fe9c63dfd.png">

b. in-progress
<img width="600" alt="download detail in-progress" src="https://user-images.githubusercontent.com/2810519/76254918-7af08700-620a-11ea-9893-7e57612214d9.png">

c. ready
<img width="600" alt="download detail ready" src="https://user-images.githubusercontent.com/2810519/76254984-9f4c6380-620a-11ea-95c7-86dd95807e9b.png">


#### Testing instructions

1. Navigate to `/backups/:siteSlug` for a site with backups enables
2. Find a valid backup ( scrolling back in time if necessary ) 
3. Note the date you are backing up to and click "Download backup'
4. Verify you are taken to `/backups/:siteSlug/donwload/:rewind` and it matches the screenshot in step 2b above.
5. Click "Create downloadable backup"
6. Verify that the page now matches the screenshot in 2c above
7. Verify that the page shortly switches to a progress bar matching 2d
8. Verify the "Download file" link works
9. Verify you separately received an email that said your download was ready
